service/src/test/java/com/epam/esm/service/CertificateServiceImplTest.java

```diff
@@ -0,0 +1,228 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dao.CertificateDAO;
+import com.epam.esm.dao.TagDAO;
+import com.epam.esm.dto.CertificateDTO;
+import com.epam.esm.dto.PatchDTO;
+import com.epam.esm.dto.QuerySpecificationDTO;
+import com.epam.esm.dto.TagDTO;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.QuerySpecification;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.exception.CertificateNotFoundException;
+import com.epam.esm.service.impl.CertificateServiceImpl;
+import com.epam.esm.util.MapperDTO;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+class CertificateServiceImplTest {
+
+    @InjectMocks
+    private CertificateServiceImpl certificateService;
+
+    @Mock
+    private CertificateDAO certificateDAO;
+
+    @Mock
+    private TagDAO tagDAO;
+
+    @Mock
+    private MapperDTO mapperDTO;
+
+    @Mock
+    private ObjectMapper objectMapper;
+
+    @Test
+    void createValid() {
+        Certificate certificate = Certificate.builder().name("test name").description("test description").price(new BigDecimal("10"))
+                .duration(5).tags(new HashSet<>()).build();
+        certificate.getTags().add(Tag.builder().name("test tag").build());
+        CertificateDTO expected = CertificateDTO.builder().name("test name").description("test description").price(new BigDecimal("10"))
+                .duration(5).tags(new HashSet<>()).build();
```

Comment:
Please move the object creation/params setting to @Before/@BeforeEach method as these test methods are difficult to read and understand what is being tested.